tests/testthat/test-config-log.R

```diff
@@ -1,6 +1,22 @@
 context("Constructing config log from bbi_config.json")
 
-check_config_ref <- function(log_df, run_nums, col_count) {
+# to minimize changes to the existing tests, we define the model and data status
+# for each of the models any particular test might need
+run_status <- dplyr::tribble(
+  ~rel_path, ~model_has_changed, ~data_has_changed,
+        "1",              FALSE,             FALSE,
+        "2",               TRUE,             FALSE,
+        "3",              FALSE,             FALSE,
+ "level2/1",               TRUE,                NA
```

Comment:
interesting... I hadn't considered this case when the data path (or model path) can't be found. It makes sense to return `NA` in that case, but I'm not actually sure that's what we want. Mainly for usability: if a user does `config_log %>% filter(!model_has_changed | !data_has_changed)` will that return the `NA` rows too? I think no, right? Maybe I'm wrong about that.

My gut is that we would want a query like this to return all the rows that had issues and missing data file is definitely an issue. Interested in your thoughts, but I think I would prefer this to return `TRUE`, or maybe more to the point, that the `file_matches` function returns `FALSE` when the file doesn't exist.

Reply:
The result of `x | y` is `NA` in the case that at least one of `x` and `y` is `NA`, and the other value is not `TRUE`. As you suggest, `filter()` does not return rows where the condition is `NA`, so a call like you describe would omit rows where `model_has_changed` or `data_has_changed` indicate an issue. In the example below, we probably want to retain rows 3, 7, and 9 from `df_or`.

```r
library(dplyr, warn.conflicts = FALSE)
df <- dplyr::tibble(
  x = rep(c(TRUE, FALSE, NA), each = 3L),
  y = rep(c(TRUE, FALSE, NA), times = 3L)
)
df_or <- dplyr::mutate(
  df, 
  x_or_y = x | y,
  not_x_or_not_y = !x | !y
)
print(df_or)
#> # A tibble: 9 x 4
#>   x     y     x_or_y not_x_or_not_y
#>   <lgl> <lgl> <lgl>  <lgl>         
#> 1 TRUE  TRUE  TRUE   FALSE         
#> 2 TRUE  FALSE TRUE   TRUE          
#> 3 TRUE  NA    TRUE   NA            
#> 4 FALSE TRUE  TRUE   TRUE          
#> 5 FALSE FALSE FALSE  TRUE          
#> 6 FALSE NA    NA     TRUE          
#> 7 NA    TRUE  TRUE   NA            
#> 8 NA    FALSE NA     TRUE          
#> 9 NA    NA    NA     NA
dplyr::filter(df_or, not_x_or_not_y)
#> # A tibble: 5 x 4
#>   x     y     x_or_y not_x_or_not_y
#>   <lgl> <lgl> <lgl>  <lgl>         
#> 1 TRUE  FALSE TRUE   TRUE          
#> 2 FALSE TRUE  TRUE   TRUE          
#> 3 FALSE FALSE FALSE  TRUE          
#> 4 FALSE NA    NA     TRUE          
#> 5 NA    FALSE NA     TRUE
```

Depending on how "serious" of a problem we think having missing files is, we could

1. have `file_matches()` return `FALSE` when the file doesn't exist

2. have `file_matches()` throw an error or raise a warning

Do you prefer one of these, or a third option?

Reply:
I think FALSE is the right move. It makes sense as is "Q: does this match? A: no there's nothing there." 

And as far as the column goes, "Q: has the data changed? A: yes, the file is gone." 

I don't think and error or warning is necessary, but like I said, I think people will try to filter on this column to see if everything is as expected and the NA rows would likely slip through the cracks.